enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/SaveButton.tsx

```diff
@@ -0,0 +1,24 @@
+import { isSavedQuestionChanged } from "metabase/query_builder/utils/question";
+import { Button } from "metabase/ui";
+import * as Lib from "metabase-lib";
+
+import { useInteractiveQuestionContext } from "../context";
+
+export const SaveButton = ({
+  onClick,
+}: {
+  onClick?: () => void;
+} = {}) => {
+  const { question, originalQuestion } = useInteractiveQuestionContext();
+
+  const canSave = question && Lib.canSave(question.query(), question.type());
```

Comment:
the logic for checking if we should reset is the same as checking if we should save. 

I think in the future we should provide utilities like hooks for these types of things - if clients want to use their own components with our logic, we can provide then with something like `isQuestionSavable` or something with a better name lol